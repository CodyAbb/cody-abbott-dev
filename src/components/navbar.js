import React from "react"
import { useStaticQuery, graphql, Link } from "gatsby"
import Img from "gatsby-image"
import style from "./navbar.module.css"

// This purenavbar is a wraper that does not contain the graphql query
// this is included so that Jest can be given dummy graphql query response
// and continue with the other tests
export const PureNavbar = ({ data, children }) => (
  <>
    <nav className={style.verticalnav}>
      <Link to="/" className={style.avatarcard}>
        <Img
          className={style.avatar}
          fixed={data.file.childImageSharp.fixed}
          alt=""
        />
        <p className={style.avatartext}>
          Cody Abbott
          <br />
          Software Developer
        </p>
      </Link>
      <Link to="/posts/" className={style.navlink}>
        <p>Posts/Tutorials</p>
      </Link>
      <Link to="/projects/" className={style.navlink}>
        <p>Projects</p>
      </Link>
    </nav>
    {children}
  </>
)

const Navbar = ({ props, children }) => {
  const data = useStaticQuery(
    graphql`
      query {
        file(relativePath: { eq: "img/facebook_profile_pic.jpg" }) {
          childImageSharp {
            fixed(width: 125, height: 125) {
              ...GatsbyImageSharpFixed
            }
          }
        }
      }
    `
  )

  return <PureNavbar {...props} children={children} data={data}></PureNavbar>
}

export default Navbar
