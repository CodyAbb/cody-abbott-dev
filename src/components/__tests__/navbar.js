import React from "react"
import renderer from "react-test-renderer"
import { PureNavbar as Navbar } from "../navbar"
describe("Navbar", () => {
  it("renders correctly", () => {
    //Created using the query from Header.js
    const data = {
      file: {
        childImageSharp: {
          fixed: {
            base64:
              "data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGQABAAMBAQAAAAAAAAAAAAAAAAECBAMF/8QAFwEAAwEAAAAAAAAAAAAAAAAAAQIDAP/aAAwDAQACEAMQAAABzavPhX2KsJ7gewnP/8QAGhAAAgMBAQAAAAAAAAAAAAAAAQIAAxEhM//aAAgBAQABBQKz04ENxSOZpjYxo7K9LhjP/8QAFhEBAQEAAAAAAAAAAAAAAAAAAAEQ/9oACAEDAQE/Aciv/8QAFhEBAQEAAAAAAAAAAAAAAAAAEAER/9oACAECAQE/AcKf/8QAHBABAAICAwEAAAAAAAAAAAAAAQARAiEQMUFR/9oACAEBAAY/ApkYen2UVwVctNxIiup3P//EAB4QAAMAAgEFAAAAAAAAAAAAAAABESExQVFhcYHh/9oACAEBAAE/IX9IidKrkN7R3o12aKc5EppRnAmVr4NHLRUl8nwf/9oADAMBAAIAAwAAABBL+D7/xAAYEQACAwAAAAAAAAAAAAAAAAAAMQEQEf/aAAgBAwEBPxDXSjSf/8QAFxEBAQEBAAAAAAAAAAAAAAAAABFBAf/aAAgBAgEBPxDC8bV//8QAHhABAAICAwADAAAAAAAAAAAAARExACFBUXFhgcH/2gAIAQEAAT8Q1JCQSVp3vGYgkR0M7w2mAi4448xNmHcV3f3kpCJq76MipBBwb/c3JIViD5Y+kXQKxj2ICnfmf//Z",
            width: 125,
            height: 125,
            src:
              "/static/3ede9b91227891490f9ae0fdf9ba25db/bac10/facebook_profile_pic.jpg",
            srcSet:
              "/static/3ede9b91227891490f9ae0fdf9ba25db/bac10/facebook_profile_pic.jpg 1x,\n/static/3ede9b91227891490f9ae0fdf9ba25db/1b59b/facebook_profile_pic.jpg 1.5x,\n/static/3ede9b91227891490f9ae0fdf9ba25db/034c8/facebook_profile_pic.jpg 2x",
          },
        },
      },
    }
    const tree = renderer.create(<Navbar data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
