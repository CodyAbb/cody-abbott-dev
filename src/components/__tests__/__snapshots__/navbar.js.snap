// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav
  className="verticalnav"
>
  <a
    className="avatarcard"
    href="/"
  >
    <div
      className="avatar gatsby-image-wrapper"
      style={
        Object {
          "display": "inline-block",
          "height": 125,
          "overflow": "hidden",
          "position": "relative",
          "width": 125,
        }
      }
    >
      <img
        alt=""
        aria-hidden={true}
        src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGQABAAMBAQAAAAAAAAAAAAAAAAECBAMF/8QAFwEAAwEAAAAAAAAAAAAAAAAAAQIDAP/aAAwDAQACEAMQAAABzavPhX2KsJ7gewnP/8QAGhAAAgMBAQAAAAAAAAAAAAAAAQIAAxEhM//aAAgBAQABBQKz04ENxSOZpjYxo7K9LhjP/8QAFhEBAQEAAAAAAAAAAAAAAAAAAAEQ/9oACAEDAQE/Aciv/8QAFhEBAQEAAAAAAAAAAAAAAAAAEAER/9oACAECAQE/AcKf/8QAHBABAAICAwEAAAAAAAAAAAAAAQARAiEQMUFR/9oACAEBAAY/ApkYen2UVwVctNxIiup3P//EAB4QAAMAAgEFAAAAAAAAAAAAAAABESExQVFhcYHh/9oACAEBAAE/IX9IidKrkN7R3o12aKc5EppRnAmVr4NHLRUl8nwf/9oADAMBAAIAAwAAABBL+D7/xAAYEQACAwAAAAAAAAAAAAAAAAAAMQEQEf/aAAgBAwEBPxDXSjSf/8QAFxEBAQEBAAAAAAAAAAAAAAAAABFBAf/aAAgBAgEBPxDC8bV//8QAHhABAAICAwADAAAAAAAAAAAAARExACFBUXFhgcH/2gAIAQEAAT8Q1JCQSVp3vGYgkR0M7w2mAi4448xNmHcV3f3kpCJq76MipBBwb/c3JIViD5Y+kXQKxj2ICnfmf//Z"
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "transitionDelay": "500ms",
            "width": "100%",
          }
        }
      />
      <picture>
        <source
          srcSet="/static/3ede9b91227891490f9ae0fdf9ba25db/bac10/facebook_profile_pic.jpg 1x,
/static/3ede9b91227891490f9ae0fdf9ba25db/1b59b/facebook_profile_pic.jpg 1.5x,
/static/3ede9b91227891490f9ae0fdf9ba25db/034c8/facebook_profile_pic.jpg 2x"
        />
        <img
          alt=""
          height={125}
          loading="lazy"
          onLoad={[Function]}
          src="/static/3ede9b91227891490f9ae0fdf9ba25db/bac10/facebook_profile_pic.jpg"
          srcSet="/static/3ede9b91227891490f9ae0fdf9ba25db/bac10/facebook_profile_pic.jpg 1x,
/static/3ede9b91227891490f9ae0fdf9ba25db/1b59b/facebook_profile_pic.jpg 1.5x,
/static/3ede9b91227891490f9ae0fdf9ba25db/034c8/facebook_profile_pic.jpg 2x"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
          width={125}
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source srcset=\\"/static/3ede9b91227891490f9ae0fdf9ba25db/bac10/facebook_profile_pic.jpg 1x,
/static/3ede9b91227891490f9ae0fdf9ba25db/1b59b/facebook_profile_pic.jpg 1.5x,
/static/3ede9b91227891490f9ae0fdf9ba25db/034c8/facebook_profile_pic.jpg 2x\\" /><img loading=\\"lazy\\" width=\\"125\\" height=\\"125\\" srcset=\\"/static/3ede9b91227891490f9ae0fdf9ba25db/bac10/facebook_profile_pic.jpg 1x,
/static/3ede9b91227891490f9ae0fdf9ba25db/1b59b/facebook_profile_pic.jpg 1.5x,
/static/3ede9b91227891490f9ae0fdf9ba25db/034c8/facebook_profile_pic.jpg 2x\\" src=\\"/static/3ede9b91227891490f9ae0fdf9ba25db/bac10/facebook_profile_pic.jpg\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
    <p
      className="avatartext"
    >
      Cody Abbott
      <br />
      Software Developer
    </p>
  </a>
  <a
    className="navlink"
    href="/posts/"
  >
    <p>
      Posts/Tutorials
    </p>
  </a>
  <a
    className="navlink"
    href="/projects/"
  >
    <p>
      Projects
    </p>
  </a>
</nav>
`;
